### PR TITLE
Georef definition enhancements

### PR DESCRIPTION
**Pull request by ataraxido**
The (technical) solution I eventually implemented in order to avoid lat/lng set on click isn't very pretty, that's the least we can say but after many attempts it's the only one I managed to have working : the code is very tangled here with many dependencies.

### CODE REVIEW

**Comment by AntoineMarnat** [comment]
To me, the fact that you can drag the items on the map but it is not taken into account, is conterintuitive, and may lead to more missbehaviour.

**Comment by brunobesson** [comment]
Could you explain more what you have done?
De-coupling the fields and the map seems a problem to me.

I also discovered that the lng and lat fields have not the `step` attribute set to `any` => could we fix it too?

**Comment by ataraxido** [comment]
I did exactly what's described here : https://github.com/c2corg/c2c_ui/issues/957#issuecomment-724992431 and seemed to be okay for everyone :)

**Comment by brunobesson** [comment]
> I did exactly what's described here

I have not followed sorry. I'll try to review your code once this behaviour is approved. Could you flag those involved to test your work?

**Comment by ataraxido** [comment]
Hello there @AntoineMarnat @cbeauchesne

**Comment by AntoineMarnat** [comment]
> I did exactly what's described here : [#957 (comment)](https://github.com/c2corg/c2c_ui/issues/957#issuecomment-724992431) and seemed to be okay for everyone :)

What I understood is that the "pick from map" button would allow to drag the georef on the map (locked map otherwise). 

I do not like the fact that what you see on the map may not be what you save, this is against intuition.

What I understood was:
two buttuns: 
- pick georef (below map next to the lat lng)
- draw on map (above map next to upload gpx)

- If neither is activated, the map is frozen: you cannot move geoloc item or gps track.
- Buttons cannot be both activated at the same time.
- If pick georef is activated, you can drag the geoloc only.
- If draw on map is activated, you can create/edit the gps track only.

I understand it might be tricky to implement.

**Comment by cbeauchesne** [comment]
> To me, the fact that you can drag the items on the map but it is not taken into account, is conterintuitive, and may lead to more missbehaviour.

What about not being able to move the picto at all?

File: src/views/wiki/edition/utils/document-edition-view-mixin.js [hunk 1]
```diff
@@ -168,19 +168,6 @@ export default {
       }
     },
 
-    setGeometryPoint() {
```
Comment (cbeauchesne) [inline src/views/wiki/edition/utils/document-edition-view-mixin.js:171]: Removing this function break the edition using numbers inputs now ?

File: src/views/wiki/edition/utils/MapInputRow.vue [hunk 2]
```diff
@@ -24,16 +24,22 @@
           show-recenter-on
         />
       </div>
-      <div class="column is-6">
+      <div class="column is-5">
         <div class="field">
           <label class="label">{{ $gettext('Longitude') }}</label>
-          <input-simple type="number" postfix="°E" v-model="longitude" @input="setGeometryPoint" />
+          <input-simple type="number" step="any" postfix="°E" v-model="longitude" @input="setGeometryPoint" />
```
Comment (cbeauchesne) [inline src/views/wiki/edition/utils/MapInputRow.vue:30]: In case of outings, we should make this inputs read only

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: src/views/wiki/edition/utils/document-edition-view-mixin.js (hunk 1 above)
Comment (ataraxido) [reply src/views/wiki/edition/utils/document-edition-view-mixin.js:171]: What do you mean exactly ? Not sure to understand, got no error and everything seems working fine

File: src/views/wiki/edition/utils/MapInputRow.vue (hunk 2 above)
Comment (ataraxido) [reply src/views/wiki/edition/utils/MapInputRow.vue:30]: So the "Pick" button shouldn't be displayed in that case, right ?

**Comment by ataraxido** [comment]
Seems what I proposed does not arrive at a consensus and in the end it's definitely not clear for me what is expected :/

**Comment by AntoineMarnat** [comment]
> What about not being able to move the picto at all?

Yes, whenever the "pick from map" button is not activated.

**Comment by asaunier** [comment]
Could you rename the PR title as "Main implementation" is not very self-explanatory about what it does (and the description does not help either) :P Thanks!

File: src/views/wiki/edition/utils/document-edition-view-mixin.js (hunk 1 above)
Comment (ataraxido) [reply src/views/wiki/edition/utils/document-edition-view-mixin.js:172]: @cbeauchesne ?

**Pull request closed without merging**